Code/EnginePlugins/AiPlugin/Navigation/Components/NavMeshObstacleComponent.cpp

```diff
@@ -0,0 +1,71 @@
+﻿#include <AiPlugin/AiPluginPCH.h>
+#include <AiPlugin/Navigation/Components/NavMeshObstacleComponent.h>
+#include <AiPlugin/Navigation/NavMesh.h>
+#include <AiPlugin/Navigation/NavMeshWorldModule.h>
+#include <Core/Interfaces/PhysicsWorldModule.h>
+#include <Core/World/World.h>
+
+// clang-format off
+EZ_BEGIN_COMPONENT_TYPE(ezNavMeshObstacleComponent, 1, ezComponentMode::Static)
+{
+  EZ_BEGIN_ATTRIBUTES
+  {
+    new ezCategoryAttribute("AI/Navigation"),
+  }
+  EZ_END_ATTRIBUTES;
+}
+EZ_END_COMPONENT_TYPE
+// clang-format on
+
+ezNavMeshObstacleComponent::ezNavMeshObstacleComponent() = default;
+ezNavMeshObstacleComponent::~ezNavMeshObstacleComponent() = default;
+
+void ezNavMeshObstacleComponent::OnActivated()
+{
+  SUPER::OnActivated();
+
+  if (IsSimulationStarted())
+    InvalidateSectors();
+}
+
+void ezNavMeshObstacleComponent::OnSimulationStarted()
+{
+  ezComponent::OnSimulationStarted();
+
+  InvalidateSectors();
+}
+
+void ezNavMeshObstacleComponent::OnDeactivated()
+{
+  InvalidateSectors();
+
+  SUPER::OnDeactivated();
+}
+
+void ezNavMeshObstacleComponent::InvalidateSectors()
+{
+  // TODO: dynamic obstacles not implemented yet
```

Comment:
So dynamic objects "don't work" because the navmesh only considers static ones, right? Though it might be made to work at least with certain kinematic objects (assuming you trigger the navmesh update yourself after e.g. a door moved). Is that what you have in mind here?

Reply:
Yeah, and I was also considering automatically tracking if an object has moved and invalidating both previous sectors and new ones if it has. So I decided to postpone it until later and do it in a separate PR (or someone else could do it). I'll investigate it tomorrow and see if we can make navmesh consider non-static objects and then after making `ezNavMeshObstacleComponent::InvalidateSectors` public it could be useful with kinematic objects